examples/lock-app/cc13x2x7_26x2x7/main/AppTask.cpp

```diff
@@ -151,6 +151,27 @@ int AppTask::Init()
             ;
     }
 
+#if CHIP_ENABLE_ADDITIONAL_DATA_ADVERTISING && CHIP_ENABLE_ROTATING_DEVICE_ID
+    {
+	uint8_t EUI64[8];
+        char stringEUI64[sizeof(EUI64) * 2 + 1]; // just big enough for string(EUI64)
+
+        // set serial number to flash
+        // this is before ZCL is started, so Basic Cluster populates Serial Number with this EUI64 (in string)
+        ThreadStackMgrImpl().GetIeeeEui64(EUI64);
+        sprintf(stringEUI64, "%02X%02X%02X%02X%02X%02X%02X%02X",
+			EUI64[0], EUI64[1], EUI64[2], EUI64[3], EUI64[4], EUI64[5], EUI64[6], EUI64[7]);
```

Comment:
Is there a reason not to use our existing BytesToHex utility here?

Reply:
@bzbarsky-apple , thanks for pointing this out, BytesToHex does sound like the perfect API to use here. I did not know that existed, will try it out and update.

Reply:
Updated here: https://github.com/project-chip/connectedhomeip/pull/15367/commits/763e4e38ae860e9f50e0b560c70b19833ef281c1